yt_idv/scene_components/base_component.py

```diff
@@ -143,11 +146,21 @@ def _default_render_method(self):
     def _change_render_method(self, change):
         new_combo = component_shaders[self.name][change["new"]]
         with self.hold_trait_notifications():
-            self.vertex_shader = new_combo["first_vertex"]
-            self.fragment_shader = new_combo["first_fragment"]
-            self.geometry_shader = new_combo.get("first_geometry", None)
-            self.colormap_vertex = new_combo["second_vertex"]
-            self.colormap_fragment = new_combo["second_fragment"]
+            self.vertex_shader = new_combo[
+                "first_vertex"
+            ], self._program1_pp_defs.get_shader_defs("vertex")
```

Comment:
tests are failing cause `self._program1_pp_defs` (and 2) can be `None`.

Reply:
ya, I do think that'd be better!

Reply:
fixed at last! added defaults for `self._program1_pp_defs` and `self._program2_pp_defs` .

Reply:
Yay!